packages/client/components/AnalyticsPage.tsx

```diff
@@ -157,7 +144,62 @@ const AnalyticsPage = () => {
       )
     }, TIME_TO_RENDER_TREE)
   }, [isSegmentLoaded, pathname])
+
   return null
 }
 
+function identifyUserWithSegment(viewerInfo?: ViewerInfo) {
+  const segmentKey = window.__ACTION__.segment
+  if (!segmentKey) return
+  if (!viewerInfo) return
+
+  const {id, email} = viewerInfo
+  safeIdentify(id, email)
+}
+
+function setupLogRocket(viewerInfo?: ViewerInfo) {
+  if (viewerInfo) {
+    const {id, email} = viewerInfo
+    LogRocketManager.setUser(id, email)
+  }
+
+  maybeRemoveIsErrorProneFlag()
+
+  const isErrorProne = getIsErrorProne()
+  const isLoggedOutAndErrorProne = !viewerInfo && isErrorProne
+  const isLoggedInAndErrorProneOrWatched = viewerInfo && (isErrorProne || viewerInfo.isWatched)
+
+  if (isLoggedOutAndErrorProne || isLoggedInAndErrorProneOrWatched) {
+    LogRocketManager.initialize()
+  }
+}
+
+function identifyUserWithDatadog(viewerInfo?: ViewerInfo) {
+  if (!datadogEnabled) return
+
+  if (viewerInfo) {
+    const {id, email, isWatched} = viewerInfo
+    datadogRum.setUser({
+      id,
+      email,
+      isWatched
+    })
+  } else {
+    datadogRum.removeUser()
+  }
+}
+
+function identifyUserWithSentry(viewerInfo?: ViewerInfo) {
```

Comment:
I moved this out of `ErrorBoundary` and into this component so everything's in one place.